src/components/NewApiDelegationPage/NewApiDelegationPage.tsx

```diff
@@ -39,20 +42,29 @@ import { CompactDeletableListItem } from '../Reusables/CompactDeletableListItem'
 import classes from './NewApiDelegationPage.module.css';
 
 export const NewApiDelegationsPage = () => {
+  const [searchString, setSearchString] = useState('');
+  const [filters, setFilters] = useState<string[]>([]);
   const delegableApis = useAppSelector((state) => state.delegableApi.presentedApiList);
   const chosenApis = useAppSelector((state) => state.delegableApi.chosenDelegableApiList);
   const chosenOrgs = useAppSelector((state) => state.delegableOrg.chosenDelegableOrgList);
   const apiProviders = useAppSelector((state) => state.delegableApi.apiProviders);
+  const loading = useAppSelector((state) => state.delegableApi.loading);
+  const error = useAppSelector((state) => state.delegableApi.error);
   const dispatch = useAppDispatch();
   const navigate = useNavigate();
-  const [searchString, setSearchString] = useState('');
-  const [filters, setFilters] = useState<string[]>([]);
   const { t } = useTranslation('common');
+  const fetchData = async () => await dispatch(fetchDelegableApis());
+
+  useEffect(() => {
+    if (loading) {
+      void fetchData();
+    }
+  }, [delegableApis.length]);
```

Comment:
I'm assuming this is here purely to cut down on the number of times the Effect should be called, and not because it should actually fetch anything every time someone chooses or removes an api.

The loading state will ofc stop the actual data fetching for all but the first time, but couldn't you assure running the Effect only happens the once by leaving the array empty? (Thus avoiding checking loading this often?)
Or are there cases where loading can go from false to true?

(Yes, this is very much a nitpick 😅)

Reply:
No this is a really good comment 😁 By doing it this way we actually reduce the number of fetches to api by 1 compared to when we don't have the array.